src/formio/validators/minMaxDatetimeValidator.js

```diff
@@ -9,9 +9,9 @@ const MinMaxDatetimeValidator = {
     const minDatetime = new Date(component.component.minDate);
     const maxDatetime = new Date(component.component.maxDate);
 
-    const errorKeys =
-      component?.openForms?.validationErrorContext?.minMaxDatetimeValidatorErrorKeys;
-    const errorMessage = errorKeys ? errorKeys[0] : 'invalidDatetime';
+    const errorKey = component?.openForms?.validationErrorContext?.minMaxDatetimeValidatorErrorKey;
+    const componentErrorMessages = component.component.errors;
+    const errorMessage = componentErrorMessages[errorKey] || errorKey;
 
     return component.t(errorMessage, {
       minDatetime: minDatetime,
```

Comment:
same pattern as above applies, however you need to change the context from `minDatetime` to `minDate` etc. If you use the same message keys, then the message templates will be the same and expect the same context.